src/rgw/rgw_op.cc

```diff
@@ -908,6 +909,25 @@ void rgw_build_iam_environment(rgw::sal::Driver* driver,
   }
 }
 
+void handle_replication_status_header(
+    const DoutPrefixProvider *dpp,
+    rgw::sal::Attrs& attrs,
+    req_state* s,
+    const ceph::real_time &obj_mtime) {
+  auto attr_iter = attrs.find(RGW_ATTR_OBJ_REPLICATION_STATUS);
+  if (attr_iter != attrs.end() && attr_iter->second.to_str() == "PENDING") {
+    if (s->object->is_sync_completed(dpp, obj_mtime)) {
+        s->object->set_atomic();
+        rgw::sal::Attrs attrs, rmattrs;
+        bufferlist bl;
+        bl.append("COMPLETED");
+        attrs[RGW_ATTR_OBJ_REPLICATION_STATUS] = bl;
+        s->object->set_obj_attrs(dpp, &attrs, &rmattrs, s->yield);
```

Comment:
hmm there's a separate bug fix for `set_obj_attrs()` in https://github.com/ceph/ceph/pull/56275 that forces peers to re-replicate, but we don't want to do here. ideally our use here would preserve the existing mtime and avoid writing a bilog/datalog entry at all

cc @smanjara

Reply:
If we don't update the object attrs then we would have to look up the bilog info every time, which doesn't seem ideal either. How should we proceed then? Open a tracker for the re-replication issue? Not sure if we would address that issue in this PR.

Reply:
PR #56275  modifies the mtime to trigger replication. we don't want that here because you only want to update the `RGW_ATTR_OBJ_REPLICATION_STATUS`. you could probably just overwrite `mtime` that I am passing into `set_attrs()`  with the `obj_mtime` from `handle_replication_status_header` to preserve the existing mtime conditional upon the `FLAG_LOG_OP` that you are passing in. does that work?

Reply:
https://github.com/ceph/ceph/pull/56275 should probably only update the mtime when FLAG_LOG_OP is set

Reply:
what can we do to unblock both of these prs?

maybe the change to add flags to set_obj_attrs() should be split into a separate commit so it can be included in both prs for now

Reply:
If we merge this PR, couldn't https://github.com/ceph/ceph/pull/56275 just be rebased on top of these changes with some small changes to adjust for the flags. that PR is pretty small

Reply:
if this pr isn't ready yet or still needs testing, then I would prefer a separate pr for setting the flag alone go in first so I can rebase on top of it.

Reply:
this pr isn't ready to merge yet, while #56275 has already been through qa. maybe it's better to get that one merged first, and make the mtime change conditional on FLAG_LOG_OP in this pr. but ideally we could find a way to avoid blocking either pr on the merge of the other

what do you think @smanjara? what's left to do before merging #56275?

Reply:
just the testcase which works with data checkpoint but not bucket checkpoint. the assert there makes sure that the sync was successful, so i don't think anything is blocking there.

Reply:
If we can fix the test in https://github.com/ceph/ceph/pull/56275 I can rebase and add that conditional logic. What is the timeline for that? These PRs have been open for 2+ months so it would be good to close them promptly

Reply:
merged #56275. please go ahead and rebase @awojno-bloomberg

Reply:
@smanjara @cbodley I rebased and updated `rgw_rados.cc` to only update mtime if log_op is true. I am not aware of a scenario where we would want to update mtime but not add a replication log, but I can update the logic if there is a use case for that. Please take a look at the update when you can. Thanks